packages/ember-glimmer/tests/integration/syntax/if-unless-test.js

```diff
@@ -24,3 +28,85 @@ moduleFor('Syntax test: {{#if}} and {{#unless}} without inverse', class extends
   }
 
 });
+
+moduleFor('Syntax test: {{#if}}', class extends RenderingTest {
+
+  ['@test child conditional should not render children if parent conditional becomes false'](assert) {
```

Comment:
I suspect this could be hoisted up to the `TogglingSyntaxConditionalsTest` level (maybe even the `TogglingConditionalsTest` if we make `foo-bar` a helper instead). Maybe something like this would work:

```javascript
let innerTemplate = this.templateFor({ cond: 'cond2', truthy: '{{foo-bar}}', falsy: '' }); // {{#if cond2}}{{foo-bar}}{{else}}{{/if}}
let outerTemplate = this.templateFor({ cond: 'cond1', truthy: innerTemplate, falsy: '' }); // {{#if cond1}}{{#if cond2}}{{foo-bar}}{{else}}{{/if}}{{else}}{{/if}}
let wrappedTemplate = this.wrapperFor([outerTemplate]);

this.render(wrappedTemplate, { cond1: true, cond2: false });

...

this.runTask(() => {
  set(this.context, 'cond2', true);
  set(this.context, 'cond1', false);
});

...
```